msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -173,16 +173,20 @@ public String getUsername() {
     @Override
     @NonNull
     public String getAuthority() {
-        // If the environment shows CIAM, we should return an authority of format https://tenant.ciamlogin.com/tenant.onmicrosoft.com
-        if (getEnvironment() != null && getEnvironment().contains(CIAMAuthority.CIAM_LOGIN_URL_SEGMENT)) {
-            // Call static method in CIAMAuthority to create the full authority uri
-            return CIAMAuthority.getFullAuthorityUrlFromAuthorityWithoutPath(getEnvironment());
-        }
         // TODO: The below logic only works for the case of AAD. We need to refactor this once we
         //  make a proper fix for B2C
         if (null != getClaims()) {
             final String iss = (String) getClaims().get("iss");
             if (!StringUtil.isEmpty(iss)) {
+                // We're using the issuer to determine whether this is a CIAM authority, as the environment
+                // be a custom-URL-domain variant, which won't contain ciamlogin.com.
+                // Note: this will work as long as the issuer for CUD doesn't change, which it will
+                // in the future. See https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2832277
+                if (iss.contains(CIAMAuthority.CIAM_LOGIN_URL_SEGMENT) && !StringUtil.isEmpty(getEnvironment())
```

Comment:
Comment is a little confusing, so we know the issuer will change eventually? Linked Item has no description

Reply:
It's highly likely that the issuer will change, yes. But the full scope and business requirements of that effort are unknown (as well as timeline), so it doesn't make sense to optimise for that now.
Indeed the ADO ticket has no description; it's been difficult to get clarity on this from the owning PM and EM. They're working on other more high priority stuff. 

Let me know if there is a better way to deal with this.